api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -2969,4 +2969,178 @@ public void saveOrder_shouldPassIfAnKnownDrugOrderForTheSameDrugFormulationSpeci
 		orderService.saveOrder(order, orderContext);
 		assertNotNull(orderService.getOrder(order.getOrderId()));
 	}
+	
+	/**
+	 * @see OrderService#getNonCodedDrugConcept()
+	 * @verifies return an null if nothing is configured
+	 */
+	@Test
+	public void getNonCodedDrugConcept_shouldReturnNullIfNothingIsConfigured() throws Exception {
+		adminService.saveGlobalProperty(new GlobalProperty(OpenmrsConstants.GP_DRUG_ORDER_DRUG_OTHER, ""));
+		assertNull(orderService.getNonCodedDrugConcept());
+	}
+	
+	/**
+	 * @see OrderService#getNonCodedDrugConcept()
+	 * @verifies return a Concept if GP is set
+	 */
+	@Test
+	public void getNonCodedDrugConcept_shouldReturnAConceptIfGPIsSet() throws Exception {
+		executeDataSet("org/openmrs/api/include/OrderServiceTest-nonCodedDrugs.xml");
+		Concept nonCodedDrugConcept = orderService.getNonCodedDrugConcept();
+		assertNotNull(nonCodedDrugConcept);
+		assertThat(nonCodedDrugConcept.getConceptId(), is(5584));
+		assertEquals(nonCodedDrugConcept.getName().getName(), "DRUG OTHER");
+		
+	}
+	
+	/**
+	 * @verifies pass if an active drug order for the same coded concept, care setting and different drug non coded exists
+	 * @see OrderService#saveOrder(Order, OrderContext)
+	 */
+	@Test
+	public void saveOrder_shouldPassForANewNonCodedDrugOrderIfAnotherActiveNonCodedDrugOrderAlreadyExists()
+	        throws Exception {
+		executeDataSet("org/openmrs/api/include/OrderServiceTest-nonCodedDrugs.xml");
+		final Patient patient = patientService.getPatient(2);
+		final Concept nonCodedConcept = conceptService.getConcept(5584);
+		//sanity check that we have an active order for the same concept
+		DrugOrder duplicateOrder = (DrugOrder) orderService.getOrder(584);
+		assertTrue(duplicateOrder.isActive());
+		assertEquals(nonCodedConcept, duplicateOrder.getConcept());
+		
+		DrugOrder drugOrder = new DrugOrder();
+		drugOrder.setPatient(patient);
```

Comment:
I think the reason this test passes could be because the new order is in a different care setting, the best way to write these tests is to copy most of the field values from the existing to ensure they are always the same